iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/database/project/DatabaseServiceBasedProjectImporter.java

```diff
@@ -75,70 +67,37 @@
 	@Override
 	public void run(PortBindings portBindings, Configuration conf,
 			Map<String, String> parameters) throws Exception {
+	    
+		Preconditions.checkArgument(parameters.containsKey(IMPORT_DATABASE_SERVICE_DBNAME), 
+                "unknown database holding projects name, required parameter '%s' is missing!", IMPORT_DATABASE_SERVICE_DBNAME);
+		
 		FileSystem fs = FileSystem.get(conf);
-		if (!parameters.containsKey(IMPORT_DATABASE_SERVICE_LOCATION)) {
-			throw new InvalidParameterException("unknown database service location, "
-					+ "required parameter '" + IMPORT_DATABASE_SERVICE_LOCATION + "' is missing!");
-		}
-		if (!parameters.containsKey(IMPORT_DATABASE_SERVICE_DBNAME)) {
-			throw new InvalidParameterException("unknown database holding projects name, "
-					+ "required parameter '" + IMPORT_DATABASE_SERVICE_DBNAME + "' is missing!");
-		}
 		
-		DataFileWriter<Project> projectWriter = null;
-		try {
-//			initializing avro datastore writer
-			projectWriter = DataStore.create(
-					new FileSystemPath(fs, portBindings.getOutput().get(PORT_OUT_PROJECT)), 
-					Project.SCHEMA$);
-//			initializing MDStore reader
-			W3CEndpointReferenceBuilder eprBuilder = new W3CEndpointReferenceBuilder();
-			eprBuilder.address(parameters.get(IMPORT_DATABASE_SERVICE_LOCATION));
-			eprBuilder.build();
-			DatabaseService databaseService = new JaxwsServiceResolverImpl().getService(
-					DatabaseService.class, eprBuilder.build());
-			Map<String, Object> requestContext = ((BindingProvider) databaseService).getRequestContext();
-			requestContext.put("javax.xml.ws.client.connectionTimeout", parameters.get(IMPORT_DATABASE_CLIENT_CONNECTION_TIMEOUT));
-			requestContext.put("javax.xml.ws.client.receiveTimeout", parameters.get(IMPORT_DATABASE_CLIENT_READ_TIMEOUT));
-			
+		try (DataFileWriter<Project> projectWriter = DataStore.create(
+		        new FileSystemPath(fs, portBindings.getOutput().get(PORT_OUT_PROJECT)), Project.SCHEMA$)) {
+
+//			initializing database reader
+			DatabaseFacade databaseFacade = ServiceFacadeUtils.instantiate(parameters);
+
 //			reading sql query content
 			StringWriter writer = new StringWriter();
-			IOUtils.copy(this.getClass().getClassLoader()
-	                .getResourceAsStream("eu/dnetlib/iis/wf/importer/database/project/sql/read_project_details_v2.sql"), 
-	                writer, "utf-8");
+			IOUtils.copy(this.getClass().getClassLoader().getResourceAsStream(queryLocation), writer, "utf-8");
```

Comment:
I would move reading this sql to private method returning already a string. 
And in this method wrap the writer in try block (it's a good convention to close opened writers - even if we know that they do nothing at closing)